See this among others:
https://github.com/dovecot/core/commit/9dbc51e77859aeb3f9870e5914b36e0dd782b242

Required so we can pick up bounces and not get rejected with
"invalid -f" because our envelope is empty.

diff --git a/src/lda/main.c b/src/lda/main.c
index 998bfdd35..869271f4f 100644
--- a/src/lda/main.c
+++ b/src/lda/main.c
@@ -244,8 +244,10 @@ static void failure_exit_callback(int *status)
 static void print_help(void)
 {
 	printf(
-"Usage: dovecot-lda [-c <config file>] [-a <address>] [-d <username>] [-p <path>]\n"
-"                   [-f <envelope sender>] [-m <mailbox>] [-e] [-k]\n");
+"Usage: dovecot-lda [-c <config file>] [-d <username>] [-p <path>]\n"
+"                   [-m <mailbox>] [-e] [-k] [-f <envelope sender>]\n"
+"                   [-a <original envelope recipient>]\n"
+"                   [-r <final envelope recipient>] \n");
 }
 
 int main(int argc, char *argv[])
@@ -311,9 +313,9 @@ int main(int argc, char *argv[])
 		case 'a':
 			/* original recipient address */
 			if (smtp_address_parse_path(ctx.pool, optarg,
-				SMTP_ADDRESS_PARSE_FLAG_ALLOW_LOCALPART |
-					SMTP_ADDRESS_PARSE_FLAG_BRACKETS_OPTIONAL,
-				&rcpt_to, &errstr) < 0) {
+			    SMTP_ADDRESS_PARSE_FLAG_ALLOW_LOCALPART |
+				SMTP_ADDRESS_PARSE_FLAG_BRACKETS_OPTIONAL,
+			    &rcpt_to, &errstr) < 0) {
 				i_fatal_status(EX_USAGE,
 					"Invalid -a parameter: %s", errstr);
 			}
@@ -330,8 +332,10 @@ int main(int argc, char *argv[])
 		case 'f':
 			/* envelope sender address */
 			if (smtp_address_parse_path(ctx.pool, optarg,
-				SMTP_ADDRESS_PARSE_FLAG_BRACKETS_OPTIONAL,
-				&mail_from, &errstr) < 0) {
+			    SMTP_ADDRESS_PARSE_FLAG_BRACKETS_OPTIONAL |
+				SMTP_ADDRESS_PARSE_FLAG_ALLOW_LOCALPART |
+				SMTP_ADDRESS_PARSE_FLAG_ALLOW_EMPTY,
+			    &mail_from, &errstr) < 0) {
 				i_fatal_status(EX_USAGE,
 					"Invalid -f parameter: %s", errstr);
 			}
@@ -358,11 +362,11 @@ int main(int argc, char *argv[])
 		case 'r':
 			/* final recipient address */
 			if (smtp_address_parse_path(ctx.pool, optarg,
-				SMTP_ADDRESS_PARSE_FLAG_ALLOW_LOCALPART |
+			    SMTP_ADDRESS_PARSE_FLAG_ALLOW_LOCALPART |
 				SMTP_ADDRESS_PARSE_FLAG_BRACKETS_OPTIONAL,
-				&final_rcpt_to, &errstr) < 0) {
+			    &final_rcpt_to, &errstr) < 0) {
 				i_fatal_status(EX_USAGE,
-					"Invalid -a parameter: %s", errstr);
+					"Invalid -r parameter: %s", errstr);
 			}
 			break;
 		default:
